go/shuffle/read.go

```diff
@@ -603,6 +603,8 @@ func walkReads(id pc.ShardID, shardSpecs []*pc.ShardSpec, shuffles []shuffle,
 			var start, stop int
 			// filtered indicates that the partition is excluded and not read by this shard.
 			var filtered = !shuffle.sourcePartitions.Matches(source.LabelSet)
+			// suspended indicates that the partition is fully suspended and has no content.
+			var suspended = source.Suspend.GetLevel() == pb.JournalSpec_Suspend_FULL
```

Comment:
I'm not sure I fully understand why we're only skipping journals that are fully suspended here. I had expected that we'd also short-circuit reads for journals that are only partially suspended after the read reaches the current write head. Is that right? Or do we do a blocking read for each partially suspended journal after reaching the current head?

Reply:
> I had expected that we'd also short-circuit reads for journals that are only partially suspended after the read reaches the current write head.

We could certainly do this, but I didn't implement it. We'll instead just use the regular read path, and have a blocking read established with that singular broker which never does anything.